### PR TITLE
Add Maori support

### PR DESCRIPTION
**Pull request by bearfriend**
[US156955](https://rally1.rallydev.com/#/?detail=/userstory/720543634719&fdp=true): Intl

Adds Maori support to date/time, numbers, and supported lang/locales. Also moves some commonly repeated lang/locale lists into Intl, which can be imported from here going forward.

More list changes to come later.

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -25,6 +27,15 @@ export const supportedLocalesDetails = [
...
 export const supportedLocales = supportedLocalesDetails.map((l) => l.code);
 
+export const frauConfig = {
+	defaultLangTag: defaultLocale,
+	primary: supportedBaseLocales,
+	regions: supportedLocalesDetails.map(l => l.code.replace(/\w\w$/, r => r.toUpperCase())),
+	aliases: {
+		'en-AU': 'en-GB'
+	}
+};
```
Comment (dlockhart) [inline lib/common.js:37]: What's this stuff for?

File: lib/common.js [hunk 2]
```diff
@@ -25,6 +27,15 @@ export const supportedLocalesDetails = [
 ];
 export const supportedLocales = supportedLocalesDetails.map((l) => l.code);
```
Comment (dlockhart) [inline lib/common.js:28]: Haha these different lists are pretty confusing and I was responsible for making some/all of them! Is it:

- `supportedLocales`: full base + regional codes... not sure what this is actually used for?
- `supportedLocalesDetails`: only used in core demo snippet
- `supportedBaseLocales`: supported locales but just the base part... seems to only be used within Intl
- `supportedLangpacks`: are these the full langpack codes coming out of Brightspace?

File: lib/common.js (hunk 1 above)
Comment (bearfriend) [reply lib/common.js:37]: This is to replace `frau-locale-provider` stuff.

File: lib/dateTime.js [hunk 3]
```diff
@@ -825,6 +826,18 @@ export function getDateTimeDescriptor() {
...
+			case 'mi':
+				dateFormats = ['dddd, d MMMM y', 'd MMMM y', 'dd-MM-y', 'MMMM yyyy', 'd MMMM', 'd MMM'];
+				months = [
+					['Kohitātea', 'Huitanguru', 'Poutūterangi', 'Paengawhāwhā', 'Haratua', 'Pipiri', 'Hōngongoi', 'Hereturikōkā', 'Mahuru', 'Whiringa-ā-nuku', 'Whiringa-ā-rangi', 'Hakihea'],
+					['Kohi', 'Hui', 'Pou', 'Pae', 'Hara', 'Pipi', 'Hōngo', 'Here', 'Mahu', 'Nuku', 'Rangi', 'Haki']
+				];
+				days = [
+					['Rātapu', 'Rāhina', 'Rātū', 'Rāapa', 'Rāpare', 'Rāmere', 'Rāhoroi'],
+					['Tap', 'Hin', 'Tū', 'Apa', 'Par', 'Mer', 'Hor'],
+					['T', 'H', 'T', 'A', 'P', 'M', 'H']
```
Comment (bearfriend) [inline lib/dateTime.js:838]: These are now confirmed correct.

File: lib/common.js (hunk 1 above)
Comment (dlockhart) [reply lib/common.js:37]: Cool. Just wondering what value it has here vs. in `frau-localize-provider` directly and having `frau-locale-provider` import `defaultLocale`, `supportedBaseLocales` and `supportedLocalesDetails` from here. Just don't love this library needing to know about "frau" or "FRAs" at all.

Comment (bearfriend) [reply lib/common.js:37]: Good point! I'll make that change.

**Review by dlockhart — APPROVED** [review]

**Pull request merged**